src/client.ts

```diff
@@ -103,11 +104,16 @@ export default function connect(
   }
 }
 
+export interface ConnectionUser {
```

Comment:
This isn't related to a DB user, right? Why did you decide to use this terminology instead of "proxy" in edgedb-python?

Reply:
I can rename to proxy. The only reason why I wrote it different is that I wanted to avoid adding complexity of defining an interface in a third common file, to reference the common object in both client.ts and pool.ts, and seemed reasonable to kept it abstract (since not only a proxy might need to cleanup resources when a connection is closed).